### PR TITLE
Bug 1819029: add a document about autoscaler alerts

### PR DESCRIPTION
**Pull request by elmiko**
This change adds the alerts.md file to the user documentation, it
describes in detail the alerts that can be generated, their causes, and
suggested solutions.

### CODE REVIEW

**Comment by elmiko** [comment]
@michaelgugino would love your feedback on this

**Comment by elmiko** [comment]
updated to make the faq link point to the top of the doc

File: docs/user/alerts.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+unschedulable pods because of this condition.
+
+### Query
+```
+# for: 20m
+cluster_autoscaler_unschedulable_pods_count{service="cluster-autoscaler-default"} > 0
+```
+
+### Possible Causes
+* The cluster autoscaler does not have enough resources to increase the size of
```
Comment (michaelgugino) [inline docs/user/alerts.md:18]: How about

"The autoscaler is unable to create new machines due to resource limits on MachineAutoscalers"

File: docs/user/alerts.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  them.
+
+### Resolution
+In many cases this alert is normal and expected depending on the configuration
+of the autoscaler. If you believe this alert is describing abnormal behavior,
+check the configuration of your ClusterAutoscaler resource to ensure that you
+have specified a large enough pool of nodes, with sufficient memory and CPU core
+values to cover the entire cluster.
+
+In rare cases it is possible that the cloud provider is taking longer than 20
```
Comment (michaelgugino) [inline docs/user/alerts.md:30]: We need to adjust this alert to longer than 20 minutes in the future.

File: docs/user/alerts.md (hunk 1 above)
Comment (elmiko) [reply docs/user/alerts.md:18]: this makes me think we need 2 entries here, the one you are proposing and one about the ClusterAutoscaler resources. i think you can hit this alert if your MachineAutoscaler doesn't have enough replicas or if your ClusterAutoscaler doesn't define enough nodes/cpu/ram. wdyt?

Comment (michaelgugino) [reply docs/user/alerts.md:18]: I concur.

**Comment by elmiko** [comment]
updated the causes and resolution for ClusterAutoscalerUnschedulablePods

**Comment by elmiko** [comment]
fixes #153

**Review by JoelSpeed — COMMENTED** [review]
Looks good, one nit, typically in docs I try to capitalise Kubernetes resource names (Pods, Machines, MachineAutoscalers, etc) to make it obvious that these are objects that we are referring to rather than generic concepts (is a "machine" referring to a Machine or a virtual machine?), not sure if others care about this as much so feel free to ignore

 /approve

**Comment by elmiko** [comment]
> Looks good, one nit, typically in docs I try to capitalise Kubernetes resource names (Pods, Machines, MachineAutoscalers, etc) to make it obvious that these are objects that we are referring to rather than generic concepts (is a "machine" referring to a Machine or a virtual machine?), not sure if others care about this as much so feel free to ignore

it's funny you mention this. i usually capitalize all the kube resource names as well, but it started to get oddly pedantic at some point so i reverted it. i'm happy to put them back though, this might be something we should look to the kube/openshift docs for inspiration (eg maybe we should enforce ourselves to always do it)

**Pull request merged**